sdk/communication/AzureCommunicationChat/Source/NotificationUtils/IdentityUtil.swift

```diff
@@ -68,3 +70,66 @@ func isGcch(id: String) -> Bool {
     let gcchAcsUserPrefix = "gcch-acs:"
     return (id.starts(with: gcchTeamsUserPrefix) || id.starts(with: gcchAcsUserPrefix))
 }
+
+/// Parses out the id/phone number portion of a user id.
+/// - Parameters:
+///   - id: The string id.
+///   - prefix: The id prefix.
+/// - Returns: The part of the id after the prefix that corresponds to the user id or phone number of a user.
+private func parse(id: String, prefix: String) -> String {
+    let index = id.index(id.startIndex, offsetBy: prefix.count)
+    return String(id.suffix(from: index))
+}
+
+/// Constructs a CommunicationIdentifier from a string id.
+/// - Parameter id: The string id.
+/// - Returns: The CommunicationIdentifier.
+internal func getIdentifier(from id: String) -> CommunicationIdentifier {
+    let publicTeamsUserPrefix = "8:orgid:"
+    let dodTeamsUserPrefix = "8:dod:"
+    let gcchTeamsUserPrefix = "8:gcch:"
+    let teamsVisitorUserPrefix = "8:teamsvisitor:"
+    let phoneNumberPrefix = "4:"
+    let acsUserPrefix = "8:acs:"
+    let spoolUserPrefix = "8:spool:"
+    let dodAcsUserPrefix = "8:dod-acs:"
+    let gcchAcsUserPrefix = "8:gcch-acs:"
+
+    if id.starts(with: publicTeamsUserPrefix) {
+        return MicrosoftTeamsUserIdentifier(
+            userId: parse(id: id, prefix: publicTeamsUserPrefix),
+            isAnonymous: false,
+            rawId: id,
+            cloudEnvironment: CommunicationCloudEnvironment.Public
+        )
+    } else if id.starts(with: dodTeamsUserPrefix) {
+        return MicrosoftTeamsUserIdentifier(
+            userId: parse(id: id, prefix: dodTeamsUserPrefix),
+            isAnonymous: false,
+            rawId: id,
+            cloudEnvironment: CommunicationCloudEnvironment.Dod
+        )
+    } else if id.starts(with: gcchTeamsUserPrefix) {
+        return MicrosoftTeamsUserIdentifier(
+            userId: parse(id: id, prefix: gcchTeamsUserPrefix),
+            isAnonymous: false,
+            rawId: id,
+            cloudEnvironment: CommunicationCloudEnvironment.Gcch
+        )
+    } else if id.starts(with: teamsVisitorUserPrefix) {
+        return MicrosoftTeamsUserIdentifier(
+            userId: parse(id: id, prefix: teamsVisitorUserPrefix),
+            isAnonymous: true
+        )
+    } else if id.starts(with: phoneNumberPrefix) {
+        return PhoneNumberIdentifier(
+            phoneNumber: parse(id: id, prefix: phoneNumberPrefix),
+            rawId: id
+        )
+    } else if id.starts(with: acsUserPrefix) || id.starts(with: spoolUserPrefix) || id
+        .starts(with: dodAcsUserPrefix) || id.starts(with: gcchAcsUserPrefix) {
+        return CommunicationUserIdentifier(id)
+    } else {
+        return UnknownIdentifier(id)
+    }
+}
```

Comment:
This will need to get replaced with https://github.com/Azure/azure-sdk-for-ios/pull/1312/files#diff-a7d2b1adee8e4923bdf6478851842c5835a9a1bcfae3b364b7563eaa48070164 which is currently blocked by architects because of some issues with enums. We have a solid concept that we now need to test.

@JoshuaLai is about verify that it works in our scenario and if so, we'll get our PR approved by ARB and merged.

Reply:
Thanks @petrsvihlik for the suggestion. I see. It would be better to reuse existing methods in Common library. 
However our team targets to release Push Notification GA on Sep 15 and it will take around 2 weeks to prepare everything before release. I discussed this with our PM and EM. We would like to wait until Aug 30 to see if this method in Common library is ready to use. If this Common PR is not merged before Aug 30, we have to go ahead with GA release. But we will still create a work item for this fix and work on it after GA release. Since it is an internal code change, we believe it will not be a breaking change. Let me know if you have any concerns. Thanks!